currency_rate_update_cmc/models/res_currency_rate_provider_CMC.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# Copyright 2018 Fork Sand Inc.
+# Copyright 2018 Ross Golder
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from collections import defaultdict
+from requests import Session
+from requests.exceptions import ConnectionError, Timeout, TooManyRedirects
+import json
+import logging
+
+from odoo import models, fields, api, _
+
+_logger = logging.getLogger(__name__)
+
+headers = {
+    'Accepts': 'application/json',
+    'X-CMC_PRO_API_KEY': 'b54bcf4d-1bca-4e8e-9a24-22ff2c3d462c',
```

Comment:
API key is shared and public?

Reply:
I am not sure. I have not dived enough in the new api of CMC. See https://pro.coinmarketcap.com/migrate/ and tell me your opinion.

Also, from all the options provided, I chose https://coinmarketcap.com/api/documentation/v1/#operation/getV1CryptocurrencyQuotesHistorical as the best option to get the needed data.

Reply:
Seems it means that we should ask end-user to provide an API key, right?
Re: API choice - I agree that it seems the best option.